### PR TITLE
Fix default Metric view name

### LINKED ISSUE #1510: Default Metric view should use instrument name.
According to the [description of Views in the Metrics SDK](https://opentelemetry.io/docs/reference/specification/metrics/sdk/#view) the name of default views should be the name of the instrument they represent.

According to the code (and unit tests), the views are using the name "otel-default-view"

### PR DESCRIPTION
**Pull request by esigo**
Fixes #1510 (issue)

## Changes
 when the view is not registered, creates and registers a new view with the provided information.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: sdk/test/metrics/view_registry_test.cc [hunk 1]
```diff
@@ -29,13 +29,17 @@ TEST(ViewRegistry, FindViewsEmptyRegistry)
...
                            count++;
-                           EXPECT_EQ(view.GetName(), "otel-default-view");
+#  if HAVE_WORKING_REGEX
+                           EXPECT_EQ(view.GetName(), "default");
                            EXPECT_EQ(view.GetDescription(), "");
+#  endif
                            EXPECT_EQ(view.GetAggregationType(), AggregationType::kDefault);
                            return true;
                          });
+#  if HAVE_WORKING_REGEX
```
Comment (lalitb) [inline sdk/test/metrics/view_registry_test.cc:39]: Oh, I forgot to create an issue to support this for gcc4.8. Will create one.

File: sdk/include/opentelemetry/sdk/metrics/view/view_registry.h [hunk 2]
```diff
@@ -46,29 +46,14 @@ class ViewRegistry
...
+    bool found = findViewsInternal(instrument_descriptor, instrumentation_scope, callback);
+    // create default view if none found;
     if (!found)
     {
-      static View view("otel-default-view");
-      if (!callback(view))
-      {
-        return false;
-      }
+      createView(instrument_descriptor, instrumentation_scope);
```
Comment (lalitb) [inline sdk/include/opentelemetry/sdk/metrics/view/view_registry.h:55]: Just thinking if we just create a static view with the empty name:
```
if (!found)
{
  static View view("");
  . . . 
}
```

And then this part of code - https://github.com/open-telemetry/opentelemetry-cpp/blob/90fd29183e36377e6c3fa878db8349c8dc7dfd88/sdk/src/metrics/meter.cc#L212 - will not overwrite the name in `InstrumentDescriptor`.

Comment (lalitb) [reply sdk/include/opentelemetry/sdk/metrics/view/view_registry.h:55]: @esigo - pinging just in case these comments got missed :)

Comment (esigo) [reply sdk/include/opentelemetry/sdk/metrics/view/view_registry.h]: @lalitb thanks, I use static view now.

File: sdk/include/opentelemetry/sdk/metrics/view/view_registry.h [hunk 3]
```diff
@@ -64,7 +64,7 @@ class ViewRegistry
     // return default view if none found;
     if (!found)
     {
-      static View view("otel-default-view");
+      static View view("");
```
Comment (ThomsonTan) [inline sdk/include/opentelemetry/sdk/metrics/view/view_registry.h:67]: Could this be declared as static const?

**Review by ThomsonTan — APPROVED** [review]

File: sdk/include/opentelemetry/sdk/metrics/view/view_registry.h (hunk 3 above)
Comment (esigo) [reply sdk/include/opentelemetry/sdk/metrics/view/view_registry.h]: thanks, yes I changed to `const`.

**Pull request merged**